tools/harness-simulation/posix/sniffer_sim/sniffer.py

```diff
@@ -161,28 +205,30 @@ def Stop(self, request, context):
         self.logger.debug('call Stop')
 
         # Validate and change the state
-        if self._state == CaptureState.NONE:
-            return sniffer_pb2.StopResponse(status=sniffer_pb2.OPERATION_ERROR, pcap_content=b'')
-        self._state = CaptureState.NONE
+        with self._state_mutex:
+            if not (self._state & CaptureState.THREAD) or not (self._state & CaptureState.SYNC):
+                return sniffer_pb2.StopResponse(status=sniffer_pb2.OPERATION_ERROR)
+            self._state = CaptureState.NONE
 
         self._thread_alive.clear()
-        self._thread.join(timeout=1)
+        self._thread.join()
         self._transport.close()
         self._pcap.close()
 
-        self._tshark_proc.terminate()
+        with self._tshark_proc_mutex:
+            self._tshark_proc.terminate()
```

Comment:
Is it necessary to use lock, since we only alter `_tshark_proc` in `Start` ?

Reply:
`self._tshark_proc.stdout.read()` in `TransferFile` will change its inner state. It should be seen as a write operation. Therefore I think the use of the lock is necessary.

Reply:
I think it may be an undefined behavior that the process terminates during someone else reading from its stdin in the other thread.